Sources/NIOFileSystem/FileSystemProtocol.swift

```diff
@@ -108,9 +108,15 @@ public protocol FileSystemProtocol: Sendable {
     /// Returns the current working directory.
     var currentWorkingDirectory: FilePath { get async throws }
 
+    /// Returns the current working directory.
+    var currentWorkingDirectoryNIO: NIOFilePath { get async throws }
+
     /// Returns the path of the temporary directory.
     var temporaryDirectory: FilePath { get async throws }
 
+    /// Returns the path of the temporary directory.
+    var temporaryDirectoryNIO: NIOFilePath { get async throws }
```

Comment:
Let's not add the `NIO` variants, we can just change the return type from `FilePath` to `NIOFilePath`.